tests/util.js

```diff
@@ -200,5 +200,57 @@ export default {
 				}
 			],
 		},
+		{
+			run: isRegisteredProperty,
+			tests: [
+				{
+					beforeAll () {
+						document.head.insertAdjacentHTML(
+							"beforeend",
+							`<link rel="stylesheet" href="./util/custom-property.css" />`,
+						);
+
+						document.head.insertAdjacentHTML(
+							"beforeend",
+							`<style>
+								@property --inline-style {
+									syntax: "<color>";
+									inherits: false;
+									initial-value: red;
+								}
+							</style>`,
+						);
+
+						let sheet = new CSSStyleSheet();
+						sheet.replaceSync(`
+							@property --adopted-stylesheet {
+								syntax: "<number>";
+								inherits: true;
+								initial-value: 1;
+							}
+						`);
+						document.adoptedStyleSheets.push(sheet);
+
+						CSS.registerProperty({
+							name: "--registered-property",
+							syntax: "small | medium | large",
+							inherits: false,
+							initialValue: "medium",
+						});
+					},
+					expect: true,
+					tests: [
+						"--external-stylesheet",
+						"--inline-style",
+						"--adopted-stylesheet",
+						"--registered-property",
```

Comment:
These tests are not atomic. Their details are defined in a general `beforeAll()`. Why? Test-specific details should be kept together with the tests. `beforeAll()` should only include stuff that is **the same across all tests**. It should not run test-specific preparatory work.

Reply:
Also, since property registration cannot be undone, these tests are not really isolated, which is why we end up having to use different property names. But the data in `tests` tells us _nothing_ about what this test does, it's basically just a description, because the _actual_ core of the test is in `beforeAll()`!

The property should not be naming the test. That's what the `name` field is for.

Ideally, the test itself should be the way the property is registered, in an iframe, and the property can then even be the same across all tests if there is nothing different about how it's registered.